1. Lectura de archivos
● Importa pandas y siguiendo las convenciones.
● Lee la base de datos y asígnela a un objeto df.
● Solicita las primeras 5 observaciones con head.

In [17]:
import pandas as pd
import numpy as np

In [21]:
df_worldcup = pd.read_csv('worldcup2014.csv')

In [23]:
df_worldcup.head()

team     continent group  group_pos  cantidad_juegos  \
0       Brazil  southamerica     A          1                3   
1       Mexico  northamerica     A          2                3   
2      Croatia        europe     A          3                3   
3     Cameroon        africa     A          4                3   
4  Netherlands        europe     B          1                3   

   juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0               2                 1                0            7   
1               2                 1                0            4   
2               1                 0                2            6   
3               0                 0                3            1   
4               3                 0                0           10   

   goles_contra goles_diferencia  puntos  clasificado  
0             2                5       7            1  
1             1                3       7            1  
2             6                0       3            0  
3             9               −8       0            0  
4             3                7       9            1

2. Frecuencia de continent: utilizando value_counts responde lo siguiente:
● ¿Cuál es el continente con una mayor presencia en la muestra?
● ¿Cuál es la probabilidad de elegir un equipo asiático al azar?
● ¿Cuál es la probabilidad de elegir un equipo africano al azar?

In [24]:
df_worldcup.continent.value_counts()

europe          13
southamerica     6
africa           5
northamerica     4
asia             4
Name: continent, dtype: int64

El continente con mayor presencia es Europa

In [25]:
print("La probabilidad de elegir un equipo asiatico es: ",(len(df_worldcup[df_worldcup.continent == 'asia'])/len(df_worldcup))*100)

La probabilidad de elegir un equipo asiatico es:  12.5


In [26]:
print("La probabilidad de elegir un equipo africano es: ",(len(df_worldcup[df_worldcup.continent == 'africa'])/len(df_worldcup))*100)

La probabilidad de elegir un equipo africano es:  15.625


3. Generación de un subconjunto de datos: por cada continente, genere un nuevo objeto que almacene sólo las observaciones del continente.
Tip: Para ello puede utilizar la siguiente sintaxis: df[df['variable'] == condicion].

In [27]:
df_wc_africa = df_worldcup[df_worldcup.continent=='africa']
df_wc_europe = df_worldcup[df_worldcup.continent=='europe']
df_wc_asia = df_worldcup[df_worldcup.continent=='asia']
df_wc_southamerica = df_worldcup[df_worldcup.continent=='southamerica']
df_wc_northamerica = df_worldcup[df_worldcup.continent=='northamerica']

4. Probabilidad de clasificación a la siguiente ronda: calcula la probabilidad de que un
país pase a la siguiente ronda (clasifique) por continente.
● ¿Cuál es la probabilidad de que un país asiático pase a la siguiente ronda?
● ¿Cuáles son los dos continentes con la mayor probabilidad de clasificar?
● ¿Cuál es la probabilidad de que un país europeo no clasifique?

In [28]:
print("la probabilidad de clasificar un equipo asiatico es: ",(len(df_wc_asia[df_wc_asia.clasificado == 1])/len(df_worldcup))*100)
print("la probabilidad de clasificar un equipo africa es: ",(len(df_wc_africa[df_wc_africa.clasificado == 1])/len(df_worldcup))*100)
print("la probabilidad de clasificar un equipo europa es: ",(len(df_wc_europe[df_wc_europe.clasificado == 1])/len(df_worldcup))*100)
print("la probabilidad de clasificar un equipo south america es: ",(len(df_wc_southamerica[df_wc_southamerica.clasificado == 1])/len(df_worldcup))*100)
print("la probabilidad de clasificar un equipo north america es: ",(len(df_wc_northamerica[df_wc_northamerica.clasificado == 1])/len(df_worldcup))*100)

la probabilidad de clasificar un equipo asiatico es:  0.0
la probabilidad de clasificar un equipo africa es:  6.25
la probabilidad de clasificar un equipo europa es:  18.75
la probabilidad de clasificar un equipo south america es:  15.625
la probabilidad de clasificar un equipo north america es:  9.375


La probabilidad de que un equipo asiatico pase a la siguiente ronda es: 0.0%
Los dos continentes con mayor probabilidad de clasificar es: Europe con un 18.75% y South America con 15,63%

In [29]:
print("La probabilidad de no clasificar un equipo europeo es: ",(len(df_wc_europe[df_wc_europe.clasificado == 0])/len(df_wc_europe))*100)

La probabilidad de no clasificar un equipo europeo es:  53.84615384615385


5. Probabilidad de victoria y derrota: se pide utilizar value_counts para calcular la probabilidad de que un continente contenga países con o sin victorias.
Para ello crea una columna adicional llamada 'ha_ganado' que contenga un 0 si es que un país no tiene juegos ganados y un 1 si es que ha ganado al menos un juego.
Sugerencias: Para crear dicha columna puede revisar:
 
 
np.where
pd.where
○ ○ ○
, de Numpy
, de Pandas
Además puede utilizar los mismos procedimientos sugeridos en el Desafío 4.
● Con base en el resultado del procedimiento recién descrito, responda lo siguiente:
A. ¿Qué continente tiene la mayor probabilidad de tener países con al menos 1 victoria?
B. ¿Qué continente tiene un nivel similar entre países que tienen o no tienen victorias?
● Reformule el código para juegos perdidos y analice los continentes con mayor probabilidad de tener países sin juegos perdidos.

In [30]:
df_worldcup['ha_ganado']= np.where(df_worldcup['juegos_ganados']>=1,1,0)

In [32]:
df_worldcup.groupby('continent')['ha_ganado'].value_counts(normalize = True)

continent     ha_ganado
africa        1            0.600000
              0            0.400000
asia          0            1.000000
europe        1            0.846154
              0            0.153846
northamerica  1            0.750000
              0            0.250000
southamerica  1            1.000000
Name: ha_ganado, dtype: float64

El continente que tiene la mayor probabilidad de tener países con al menos 1 victoria, es Southamerica con 100%
El continente que tiene un nivel similar entre países que tienen o no tienen victorias, es Africa con un 60% de Victorias y un 40% de derrotas

In [33]:
df_worldcup['ha_perdido']= np.where(df_worldcup['juegos_perdidos']>=1,1,0)

In [34]:
df_worldcup.groupby('continent')['ha_perdido'].value_counts(normalize = True)

continent     ha_perdido
africa        1             1.000000
asia          1             1.000000
europe        1             0.692308
              0             0.307692
northamerica  0             0.500000
              1             0.500000
southamerica  0             0.500000
              1             0.500000
Name: ha_perdido, dtype: float64

El continente que tiene la mayor probabilidad de tener países con menos derrotas, es Europe con 30,7%